bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/RuleRegistryImpl.java

```diff
@@ -71,13 +89,15 @@ private String addIntoRuleEngine(Rule element) {
 
     @Override
     public synchronized Rule remove(String key) {
+        notify(RuleEventFactory.createRuleRemovedEvent(get(key), SOURCE));
```

Comment:
This way the event is fired, before it is removed. Please also use "notifyListenerAboutRemovedElement". See ThingRegistry for example.